reportportal_client/items/rp_base_item.py

```diff
@@ -0,0 +1,99 @@
+"""
+This module contains functional for Base RP items management.
+
+Copyright (c) 2018 http://reportportal.io .
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from reportportal_client.core.rp_requests import HttpRequest
+from reportportal_client.core.rp_responses import RPResponse
+from reportportal_client.static.defines import NOT_FOUND
+
+
+class BaseRPItem(object):
+    """This model stores attributes related to RP item."""
+
+    def __init__(self, rp_url, session, api_version, project_name,
+                 launch_uuid, generated_id):
+        """Initialize instance attributes.
+
+        :param rp_url:         report portal url
+        :param session:        Session object
+        :param api_version:    RP API version
+        :param project_name:   RP project name
+        :param launch_uuid:    Parent launch UUID
+        :param generated_id:   Id generated to speed up client
+        """
+        self.uuid = None
+        self.weight = None
+        self.generated_id = generated_id
+        self.http_requests = list()
+        self.responses = []
```

Comment:
different list styles

Reply:
It's better to use brackets instead of the function call.

Reply:
Fixed